sql/sql_optimizer.cc

```diff
@@ -765,7 +765,8 @@ JOIN::optimize()
   if (!(select_options & SELECT_DESCRIBE) &&
       !select_lex->materialized_table_count)
   {
-    init_ftfuncs(thd, select_lex, order);
+    if (init_ftfuncs(thd, select_lex, order))
```

Comment:
They were. This was fixed in 5.7 by bug fix:21140111. I have ported this bug into 5.6. The bug fix is not completely correct - needed to move check for errors from init_ftfuncs bit higher in sql_delete.cc, otherwise error from init_ftfuncs caused assertion problem.